aiida_mlip/calculations/train.py

```diff
@@ -0,0 +1,197 @@
+
+        # Define inputs
+        spec.input(
+            "mlip_config",
+            valid_type=JanusConfigfile,
+            required=True,
+            help="Config file with parameters for training",
+        )
+        spec.input(
+            "metadata.options.output_filename",
+            valid_type=str,
+            default=cls.DEFAULT_OUTPUT_FILE,
+        )
+
+        spec.input(
+            "metadata.options.scheduler_stdout",
+            valid_type=str,
+            default="_scheduler-stdout.txt",
+            help="Filename to which the content of stdout of the scheduler is written.",
+        )
+        spec.inputs["metadata"]["options"]["parser_name"].default = "janus.train_parser"
+        spec.inputs.validator = validate_inputs
+        spec.output("model", valid_type=ModelData)
+        spec.output("compiled_model", valid_type=SinglefileData)
+        spec.output(
+            "results_dict",
+            valid_type=Dict,
+            help="The `results_dict` output node of the training.",
+        )
+        spec.output("logs", valid_type=FolderData)
+        spec.output("checkpoints", valid_type=FolderData)
+        spec.default_output_node = "results_dict"
+        # Exit codes
+        spec.exit_code(
+            305,
+            "ERROR_MISSING_OUTPUT_FILES",
+            message="Some output files missing or cannot be read",
+        )
+
+    # pylint: disable=too-many-locals
+    def prepare_for_submission(
+        self, folder: aiida.common.folders.Folder
+    ) -> datastructures.CalcInfo:
+        """
+        Create the input files for the `Calcjob`.
+
+        Parameters
+        ----------
+        folder : aiida.common.folders.Folder
+            An `aiida.common.folders.Folder` to temporarily write files on disk.
+
+        Returns
+        -------
+        aiida.common.datastructures.CalcInfo
+            An instance of `aiida.common.datastructures.CalcInfo`.
+        """
+        # The config file needs to be copied in the working folder
+        # Read content
+        mlip_dict = self.inputs.mlip_config.as_dictionary
+        config_parse = self.inputs.mlip_config.get_content()
+        # Extract paths from the config
+        abs_train_folder = (Path(mlip_dict["train_file"])).resolve()
+
+        abs_test_folder = (Path(mlip_dict["test_file"])).resolve()
+
+        abs_valid_folder = (Path(mlip_dict["valid_file"])).resolve()
+        # Update the config file with absolute paths
+        config_parse = config_parse.replace(
+            mlip_dict["train_file"], str(abs_train_folder)
+        )
+        config_parse = config_parse.replace(
+            mlip_dict["test_file"], str(abs_test_folder)
+        )
+        config_parse = config_parse.replace(
+            mlip_dict["valid_file"], str(abs_valid_folder)
+        )
+
+        # Copy config file content inside the folder where the calculation is run
+        with folder.open("mlip_train.yml", "w", encoding="utf-8") as configfile:
```

Comment:
Does this file need to be an input?

Reply:
I think it's nicer to have it in the folder, like easier to find and have an immediate idea of what's in the folder where the calculation is run, but it doesn't need to I guess

Reply:
Sorry, I meant does the name of `mlip_train.yml` need to be a variable? Sorry for being unclear. Writing in a hurry and knowing what I meant.

Reply:
do you mean if the user can change it and then it should be a variable that takes this into account? in this case no, this is just the name of a copied file that I want to be always the same.
if instead you mean to save it into a variable rather then passing the string every time I use it (which is twice in this code), I changed it now